### PR TITLE
rename EndpointsClient to LocalClient

### LINKED ISSUE #484: comprehensive comments/docs on the Endpoints client and the kpng/client/client.go
The *EndpointsClient* is a little confusing to read.  

- I THINK It doesnt specifically filter on endpoints, and in any case, we have endpointslices, services, and endpoints, which i believe it filters and watches on.  

- Also it doesnt say much about how the watch mechanism works and how its node specific

```
// doesnt EndpointsClient watch everything ? it sais its just a client to the endpoints API...
// EndpointsClient is a simple client to kube-proxy's Endpoints API.
type EndpointsClient struct {
	// Target is the gRPC dial target
	Target string
```

### PR DESCRIPTION
**Pull request by abhaikollara**
### What kind of PR is this?
Improvement

### Why this PR is needed / What this PR do?
See https://github.com/kubernetes-sigs/kpng/issues/484#issuecomment-1522807182. The PR renames the `EndpointsClient` to `LocalClient` as per the suggested comment

### Which issue(s) this PR fixes?

Fixes #484

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -26,9 +26,6 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
-	// allow multi gRPC URLs
-	//_ "github.com/Jille/grpc-multi-resolver"
```
Comment (daman1807) [inline client/client.go:30]: are you sure we don't need any initialization to happen here?

Comment (abhaikollara) [reply client/client.go:30]: The code was already commented out, was just cleaning it up. 

I'm guessing, even if the init is required, it should be part of another PR ?

Comment (daman1807) [reply client/client.go:30]: oh, let it be as it is then :)

Comment (abhaikollara) [reply client/client.go]: Have reverted my change

**Comment by daman1807** [comment]
@abhaikollara can you squash the two commits

**Comment by abhaikollara** [comment]
> @abhaikollara can you squash the two commits

Done

**Comment by daman1807** [comment]
/lgtm
thanks !

**Comment by daman1807** [comment]
deferring to @knabben for final approval, I don't have much context here.

**Comment by mcluseau** [comment]
that will make @jayunit100  happy ;)
/approve

**Pull request merged**